src/core/timer.c

```diff
@@ -332,7 +332,16 @@ static void add_random(Timer *t, usec_t *v) {
         if (*v == USEC_INFINITY)
                 return;
 
-        add = random_u64() % t->random_usec;
+        if (t->randomize_first_run_only) {
+                if (t->random_value_usec == USEC_INFINITY) {
+                        t->random_value_usec = random_u64() % t->random_usec;
+                        if (t->random_value_usec == USEC_INFINITY)
+                                t->random_value_usec -= 1; /* Highest possible value, that is not USEC_INFINITY */
```

Comment:
write this as `(usec_t) -2` please, i.e. make this explicit.

Reply:
but, how can this happen? i.e. `t->random_sec` can be USEC_INFINITY at most, which means `t->random-value_sec` would be < USEC_INFINITY anyway. These two lines hence appear entirely unnecessary